provider/gandi/gandi.go

```diff
@@ -48,15 +49,49 @@ type GandiProvider struct {
 	LiveDNSClient LiveDNSClientAdapter
 	DomainClient  DomainClientAdapter
 	domainFilter  endpoint.DomainFilter
+	PreferALIAS   bool
 	DryRun        bool
+	TXTPrefix     string
+	TXTSuffix     string
 }
 
-func NewGandiProvider(ctx context.Context, domainFilter endpoint.DomainFilter, dryRun bool) (*GandiProvider, error) {
+const (
+	// RecordTypeALIAS is a RecordType enum value
+	RecordTypeALIAS = "ALIAS"
+)
+
+// inferZone determines the zone based on the RrsetName
+func inferZone(RrsetName string, TXTPrefix string, TXTSuffix string) string {
+	cleanRrsetName := strings.Replace(RrsetName, TXTPrefix, "", 1)
+	cleanRrsetName = strings.Replace(cleanRrsetName, TXTSuffix, "", 1)
+	cleanRrsetName = strings.Replace(cleanRrsetName, "cname-", "", 1)
+
+	return cleanRrsetName
+}
+
+// useAlias determines if the provider specific alias should be set
+func useAlias(domain string, preferALIAS bool) bool {
+	if preferALIAS {
+		return govalidator.IsDNSName(domain)
+	}
+
+	return false
+}
+
+// isGandiAlias determines if a given endpoint is supposed to create an Gandi Alias record
+func isGandiAlias(r livedns.DomainRecord, preferALIAS bool) bool {
+	return preferALIAS && r.RrsetType == RecordTypeALIAS
```

Comment:
Why does the setting of `preferAlias` affect whether an existing `DomainRecord` is an alias?

Reply:
Since we removed the "prefer alias" mechanism, I think this is no longer an issue?